### PR TITLE
Kafka upgrade to julia version 1.1

### PR DESCRIPTION
**Pull request by atdhcs**

### CODE REVIEW

**Comment by atdhcs** [comment]
commented finalizer function in api.jl as it get errored out...

julia> p = KafkaProducer("node1:9092")
ERROR: objects of type getfield(RDKafka, Symbol("##1#2")){Ptr{Nothing}} cannot be finalized
Stacktrace:
[1] error(::String, ::Type, ::String) at ./error.jl:42
[2] finalizer(::Any, ::Any) at ./gcutils.jl:16
[3] KafkaClient(::Int32, ::Dict{Any,Any}) at /home/csubrama/.julia/dev/RDKafka/src/api.jl:29
[4] KafkaProducer(::Dict{Any,Any}) at /home/csubrama/.julia/dev/RDKafka/src/api.jl:68
[5] KafkaProducer(::String, ::Dict{Any,Any}) at /home/csubrama/.julia/dev/RDKafka/src/api.jl:75
[6] KafkaProducer(::String) at /home/csubrama/.julia/dev/RDKafka/src/api.jl:74
[7] top-level scope at none:0

error is thrown because of mutable object looks like.

File: deps/build.log [hunk 1]
```diff
@@ -0,0 +1,11 @@
+ERROR: LoadError: ArgumentError: Package HTTP not found in current path:
```
Comment (dfdx) [inline deps/build.log]: I think this file can be added to `.gitignore`

**Comment by dfdx** [comment]
Believe it or not, but the problem with finalizer is caused simply by wrong argument order. Perhaps in Julia < 1.0 both signatures `finializer(f, x)` and `finalizer(x, f)` were supported, and Julia 1.x removed the last one for consistency. So the fix is as easy as:

```
finalizer(client -> kafka_destroy(rk), client)
```
and same for other finalizers. Do you want to make these changes yourself and finish PR or should I better create an alternative one with my changes? 

By the way, the recommended way to migrate from Julia 0.6 to 1.x is via 0.7 which is "1.0 with deprecation warnings" - usually this path uncovers most of the issues.

File: deps/build.log (hunk 1 above)
Comment (atdhcs) [reply deps/build.log]: ok sure. will make the recommended changes including finalizer.

**Comment by atdhcs** [comment]
will make the recommended changes including finalizer. I just want to let you know that still i couldn't send successful message to broker yet though I don't see any failures (i got return value as zero from ccall).
p = KafkaProducer("localhost:9092")
partition = 0
produce(p, "mytopic", partition, "message key", "message payload")

I realized i'm still working on old wrapper.jl but the .so object is made based on latest clone from RDKafka. will that be a problem? I  created new wrapper clang.jl. should i put my focus on replacing old wrapper.jl to new or can i debug with old wrapper.jl? if i want to debug what tricks do we have?

**Comment by dfdx** [comment]
> I just want to let you know that still i couldn't send successful message to broker yet though I don't see any failures (i got return value as zero from ccall).

Zero return status only tells librdkafka that the message has been sent. To find out delivery status we need to set up a [callback](https://github.com/edenhill/librdkafka/blob/master/src/rdkafka.h#L1557) for produced messages, but it requires additional development and it would be unfair to put it on you. So if don't succeed in debugging the issue with `produce()` in the nearest time, I'll try to finish it myself.

One important thing to check before giving up is that Kafka itself is set up correctly, in particular: 

1. Zookeeper is running.
2. Kafka broker is running.
3. You can send and receive messages using built-in console consumer (https://kafka.apache.org/quickstart#quickstart_send).
4. Topic name and partition are specified correctly in Julia code. 

> I realized i'm still working on old wrapper.jl but the .so object is made based on latest clone from RDKafka. 

This should not be a problem as long as librdkafka doesn't change its API which is unlikely. 

> I created new wrapper clang.jl. should i put my focus on replacing old wrapper.jl to new or can i debug with old wrapper.jl?

As far as I understand, you generated clang.jl automatically, but original wrapper.jl was created (mostly) manually and simplifies some aspects of C API. Generated wrapper may give better API coverage, so I'm open to this change, but perhaps it's better to implement it in a separate PR to keeps modifications atomic.

**Comment by atdhcs** [comment]
_One important thing to check before giving up is that Kafka itself is set up correctly, in particular:
Zookeeper is running.
Kafka broker is running.
You can send and receive messages using built-in console consumer (https://kafka.apache.org/quickstart#quickstart_send).
Topic name and partition are specified correctly in Julia code._  I tested Kafka working well with pykafka and build in tool too.
     _I'm open to this change, but perhaps it's better to implement it in a separate PR to keeps modifications atomic_ ok thanks and it make sense.

_To find out delivery status we need to set up a callback for produced messages, but it requires additional development and it would be unfair to put it on you_ 
 - I started using pykafka calling from julia as a workaround but i'm happy to help. But will have to find time to do it.

**Comment by dfdx** [comment]
Got it, thanks! I'll then take your code, try to fix the issue with producer and come back to you.

**Comment by dfdx** [comment]
I've fixed the issue with producer in #2. Waiting for CI to finish its job, but I think we are ready to use RDKafka from Julia 1.x.

**Pull request merged**